includes/gateways/stripe/includes/payment-methods/class-give-stripe-sepa.php

```diff
@@ -0,0 +1,347 @@
+				$stripe_customer_id   = $give_stripe_customer->get_id();
+				$payment_method_id    = ! empty( $give_stripe_customer->attached_payment_method ) ?
+					$give_stripe_customer->attached_payment_method->id :
+					$payment_method;
+
+				// We have a Stripe customer, charge them.
+				if ( $stripe_customer_id ) {
+
+					// Setup the payment details.
+					$payment_data = array(
+						'price'           => $donation_data['price'],
+						'give_form_title' => $donation_data['post_data']['give-form-title'],
+						'give_form_id'    => $form_id,
+						'give_price_id'   => $price_id,
+						'date'            => $donation_data['date'],
+						'user_email'      => $donation_data['user_email'],
+						'purchase_key'    => $donation_data['purchase_key'],
+						'currency'        => give_get_currency( $form_id ),
+						'user_info'       => $donation_data['user_info'],
+						'status'          => 'pending',
+						'gateway'         => $this->id,
+					);
+
+					// Record the pending payment in Give.
+					$donation_id = give_insert_payment( $payment_data );
+
+					// Return error, if donation id doesn't exists.
+					if ( ! $donation_id ) {
+						give_record_gateway_error(
+							__( 'Donation creating error', 'give' ),
+							sprintf(
+								/* translators: %s Donation Data */
+								__( 'Unable to create a pending donation. Details: %s', 'give' ),
+								wp_json_encode( $donation_data )
+							)
+						);
+						give_set_error( 'stripe_error', __( 'The Stripe Gateway returned an error while creating a pending donation.', 'give' ) );
+						give_send_back_to_checkout( '?payment-mode=' . give_clean( $_GET['payment-mode'] ) );
+						return;
+					}
+
+					// Assign required data to array of donation data for future reference.
+					$donation_data['donation_id'] = $donation_id;
+					$donation_data['description'] = $donation_summary;
+					$donation_data['customer_id'] = $stripe_customer_id;
+					$donation_data['source_id']   = $payment_method;
+
+					// Save Stripe Customer ID to Donation note, Donor and Donation for future reference.
+					give_insert_payment_note( $donation_id, 'Stripe Customer ID: ' . $stripe_customer_id );
+					$this->save_stripe_customer_id( $stripe_customer_id, $donation_id );
+					give_update_meta( $donation_id, '_give_stripe_customer_id', $stripe_customer_id );
+
+					// Save Source ID to donation note and DB.
+					give_insert_payment_note( $donation_id, 'Stripe Source/Payment Method ID: ' . $payment_method_id );
+					give_update_meta( $donation_id, '_give_stripe_source_id', $payment_method_id );
+					give_update_meta( $donation_id, '_give_stripe_payment_method_id', $payment_method_id );
+
+					// Save donation summary to donation.
+					give_update_meta( $donation_id, '_give_stripe_donation_summary', $donation_summary );
+
+					/**
+					 * This filter hook is used to update the payment intent arguments.
+					 *
+					 * @since 2.5.0
+					 */
+					$intent_args = apply_filters(
```

Comment:
@mehul0810 where are we adding / removing the fee based on whether the user is on Stripe free or Premium?